app/helpers/mailer_helper.rb

```diff
@@ -215,4 +309,26 @@ def style_work_tag_metadata_list(tags)
       work_tag_metadata_list(tags)
     end
   end
+
+  def batch_subscription_preface_key(creation, email_format:)
+    base_key = "user_mailer.batch_subscription_notification.preface"
+    creator_key = creation.anonymous? ? "anon" : "named"
+    creation_key = creation.is_a?(Chapter) ? "chapter" : "work"
+    dating_key = creation.backdate ? ".backdated" : ".new" if creation.is_a?(Work)
+    format_key = ".#{email_format}" unless creation.is_a?(Work) && creation.anonymous?
+
+    # Note the lack of . before dating and format keys, which are not always
+    # included.
```

Comment:
Rather than worrying about when to insert a `.`, would it maybe make sense to construct the key as an array and join it at the end? e.g.
```ruby
key = [:user_mailer, :batch_subscription_notification, :preface]
key << (creation.anonymous? ? :anon : :named)
key << creation.model_name.i18n_key
key << (creation.backdate ? :backdated : :new) if creation.is_a?(Work)
key << email_format unless creation.is_a?(Work) && creation.anonymous?
key.join(".")